src/net.cpp

```diff
@@ -2930,24 +2930,19 @@ void CConnman::PushMessage(CNode* pnode, CSerializedNetMsg&& msg)
     size_t nBytesSent = 0;
     {
         LOCK(pnode->cs_vSend);
-        bool optimisticSend(pnode->vSendMsg.empty());
 
         //log total amount of bytes per message type
         pnode->mapSendBytesPerMsgCmd[msg.m_type] += nTotalSize;
         pnode->nSendSize += nTotalSize;
 
-        if (pnode->nSendSize > nSendBufferMaxSize)
-            pnode->fPauseSend = true;
+        if (pnode->nSendSize > nSendBufferMaxSize) pnode->fPauseSend = true;
         pnode->vSendMsg.push_back(std::move(serializedHeader));
-        if (nMessageSize)
-            pnode->vSendMsg.push_back(std::move(msg.data));
+        if (nMessageSize) pnode->vSendMsg.push_back(std::move(msg.data));
 
         // If write queue empty, attempt "optimistic write"
-        if (optimisticSend == true)
-            nBytesSent = SocketSendData(pnode);
+        if (pnode->vSendMsg.empty()) nBytesSent = SocketSendData(pnode);
```

Comment:
Any reason for this behavior change?

Reply:
Probably causing the ci failures

Reply:
c2f62aefabc2b00ce0ed28901d80d636b600feef

Exactly, msg.data is pushed to `vSendMsg` right above.

Reply:
Not a behavior change!  Just removed the optimisticSend boolean.  Suggested here https://github.com/bitcoin/bitcoin/pull/19509#discussion_r481094927

Reply:
This is not a behavior change.  It just removes the unused boolean.  It's done in this small cleanup commit c2f62aefabc2b00ce0ed28901d80d636b600feef.  Still checking the ci failure.

Reply:
It was suggested here https://github.com/bitcoin/bitcoin/pull/19509#discussion_r481094927

Reply:
Yep this was silly of me, thanks for finding the bug!